jenkins/release-workflows/release-notes-generate.jenkinsfile

```diff
@@ -0,0 +1,171 @@
+                                                        println("Filename is ${filename}")
+                                                        println("Component object: ${componentObj.name}, ${componentObj.ref}, ${componentObj.repository}")
+                                                        echo "Current workspace is $WORKSPACE}"
+                                                        withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+                                                            sh """
+                                                                #!/bin/bash
+                                                                set +e
+                                                                ./release_notes.sh generate manifests/${INPUT_MANIFEST} --component ${currentComponent} --date ${GIT_LOG_DATE} --max-tokens ${MAX_TOKENS} --ref ${REF}
+
+                                                                if [ \$? -ne 0 ]; then
+                                                                    echo "Release notes generation failed for component ${currentComponent}"
+                                                                    exit 1
+                                                                else
+                                                                    COMPONENT_REPO_URL="${componentObj.repository}"
+                                                                    REPO_NAME=\$(basename "\${COMPONENT_REPO_URL}" .git)
+
+                                                                    git clone -b ${componentObj.ref} "\${COMPONENT_REPO_URL}"
+                                                                    cd "\${REPO_NAME}"
+
+                                                                    if [ -f "$WORKSPACE/release-notes/${filename}" ]; then
+                                                                        cp "$WORKSPACE/release-notes/${filename}" release-notes/
+                                                                        if [ -z "\$(git status --porcelain)" ]; then
+                                                                             echo "No changes to commit for component \${REPO_NAME}. Skipping PR creation."
+                                                                        else
+                                                                            git remote set-url origin "https://opensearch-ci:${GITHUB_TOKEN}@github.com/opensearch-project/\$REPO_NAME"
+                                                                            git config user.email "opensearch-infra@amazon.com"
+                                                                            git config user.name "opensearch-ci"
+                                                                            git checkout -b release/release-notes-${version}
```

Comment:
I dont think its a good idea to create new branch for each version. They would be kept lingering. How about always push to `releaseNotes` branch? Hard reset to base branch everytime and then proceed?
We do the same for `update-manifest` and `create-pull-request` which is for version increments. https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-increment-plugin-versions.yml#L161